src/runtime/Story.svelte

```diff
@@ -106,17 +142,27 @@
 </script>
 
 {#if isCurrentlyViewed}
-  {#if children}
-    {@render children(renderer.args, renderer.storyContext)}
-  {:else if template}
+  {#if template && isSnippet(template)}
     {@render template(renderer.args, renderer.storyContext)}
+  {:else if children && isSnippet(children)}
```

Comment:
```suggestion
  {:else if isSnippet(children)}
```

isSnippet function can deal with undefined